docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -95,24 +95,72 @@ several functions accessible with the naming scheme
 ``process_output_<output-format>``.
 
 - ``process_output_regex``: Extracts results using a regex.
+
+  ::
+
+      @linter(executable='my_tool',
+              use_stderr=True)
+      class MyBear:
+          # Assuming the tool puts some issue messages into stderr.
+          def process_output(self, output, filename, file):
+              # output is a string, as we activated just ``use_stderr``
+              map = {'info': RESULT_SEVERITY.INFO,
+                     'warn': RESULT_SEVERITY.NORMAL,
+                     'error': RESULT_SEVERITY.MAJOR}
+              regex = "(?P<line>\d+):(?P<message>(?P<severity>[WEI]).*)"
+              yield from self.process_output_regex(
+                                  stderr, filename, file, regex, map)
+
+  A static message to use for results instead of grabbing it from the executable
+  output (via the ``message`` named regex group) can also be provided using the
+  ``result_message`` parameter.
+
 - ``process_output_corrected``: Extracts results (with patches) by using a
   corrected version of the file processed.
 
-::
+  ::
 
     @linter(executable='my_tool',
-            use_stdout=True,
-            use_stderr=True)
+            use_stdout=True)
     class MyBear:
-        # Assuming the tool puts a corrected version of the file into stdout
-        # and additional issue messages (that can't be fixed automatically)
-        # into stderr, let's combine both streams!
-        def process_output(self, output, filename, file):
-            # output is now a tuple, as we activated both, stdout and stderr.
-            stdout, stderr = output
-            yield from self.process_output_corrected(stdout, filename, file)
-            regex = "(?P<message>.*)"
-            yield from self.process_output_regex(stderr, filename, file, regex)
+        # Assuming the tool puts a corrected version of the file into stdout.
+        def process_output:
+            # output is a string, as we activated just ``use_stdout``
+            yield from self.process_output_corrected(
+                                stdout, filename, file,
+                                diff_severity=RESULT_SEVERITY.NORMAL,
+                                diff_distance=2)
+
+  Here, the ``diff_distance`` parameter takes the number of unchanged lines
+  allowed in between two changed lines so they get yielded as a single diff.
+  If a negative distance is given, every change will be yielded as an own
```

Comment:
a difference of `0` would have the same effect, and be much more sensible?

Reply:
i.e. providing a negative distance == providing 0, right?
>If a negative distance is given, every change will be yielded as an own.

This was mentioned in the docstring I guess, I took it from there.

Reply:
Can you investigate if 0 is the same as a negative number?
If so, the docstrings also need to be fixed, encouraging use of `0` instead of any random negative number.

Reply:
>if 0 is the same as a negative number?
No, according to https://github.com/coala/coala-bears/pull/2335#discussion_r179727839. So, docs are fine. :)